sockeye/encoder.py

```diff
@@ -108,6 +131,18 @@ def get_rnn_cells(self) -> List[mx.rnn.BaseRNNCell]:
         """
         raise NotImplementedError()
 
+    def get_encoded_data_length(self, data_length: mx.sym.Symbol) -> mx.sym.Symbol:
+        """
+        Returns the data lengths of the encoded sequence.
+        """
+        return data_length
+
+    def get_encoded_seq_len(self, seq_len: int) -> int:
+        """
+        Returns the size of the encoded sequence.
+        """
+        return seq_len
+
```

Comment:
Encoders now report modifications to input sequence length.  Encoders that don't modify length don't need to implement these methods as the default is just returning the input length.


 
--



 
--